pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -473,6 +482,17 @@ func (n *NodeInfo) Clone() *NodeInfo {
 // VolumeLimits returns volume limits associated with the node
 func (n *NodeInfo) VolumeLimits() map[v1.ResourceName]int64 {
 	volumeLimits := map[v1.ResourceName]int64{}
+
+	if n.csiNode != nil {
```

Comment:
IMO, if CSINode exists and it contains the driver, we should use it and not to fall back to `node.status.allocatable`. Should be the logic reversed, collect `node.allocatable` first and overwrite it with `csiNode.*.allocatable` later?